--- src/Terraria/Terraria/GameContent/UI/States/UIWorkshopHub.cs
+++ src/BetterWorkshop/Terraria/GameContent/UI/States/UIWorkshopHub.cs
@@ -20,6 +_,7 @@
 		private UIElement _buttonPublishWorlds;
 		private UIElement _buttonBack;
 		private UIElement _buttonLogs;
+		private UIElement _buttonProfile;
 		private UIGamepadHelper _helper;
 
 		public static event Action OnWorkshopHubMenuOpened;
@@ -59,7 +_,7 @@
 			uITextPanel.SetPadding(15f);
 			uITextPanel.BackgroundColor = new Color(73, 94, 171);
 			UITextPanel<LocalizedText> uITextPanel2 = new UITextPanel<LocalizedText>(Language.GetText("UI.Back"), 0.7f, large: true);
-			uITextPanel2.Width.Set(-10f, 0.5f);
+			uITextPanel2.Width.Set(-10f, 1f / 3f);
 			uITextPanel2.Height.Set(50f, 0f);
 			uITextPanel2.VAlign = 1f;
 			uITextPanel2.HAlign = 0f;
@@ -71,10 +_,10 @@
 			uIElement.Append(uITextPanel2);
 			_buttonBack = uITextPanel2;
 			UITextPanel<LocalizedText> uITextPanel3 = new UITextPanel<LocalizedText>(Language.GetText("Workshop.ReportLogsButton"), 0.7f, large: true);
-			uITextPanel3.Width.Set(-10f, 0.5f);
+			uITextPanel3.Width.Set(-10f, 1f / 3f);
 			uITextPanel3.Height.Set(50f, 0f);
 			uITextPanel3.VAlign = 1f;
-			uITextPanel3.HAlign = 1f;
+			uITextPanel3.HAlign = 0.5f;
 			uITextPanel3.Top.Set(-num, 0f);
 			uITextPanel3.OnMouseOver += FadedMouseOver;
 			uITextPanel3.OnMouseOut += FadedMouseOut;
@@ -82,6 +_,18 @@
 			uITextPanel3.SetSnapPoint("Logs", 0);
 			uIElement.Append(uITextPanel3);
 			_buttonLogs = uITextPanel3;
+			UITextPanel<LocalizedText> profileButton = new UITextPanel<LocalizedText>(Language.GetText("BetterWorkshop.ProfileButton"), 0.7f, large: true);
+			profileButton.Width.Set(-10f, 1f / 3f);
+			profileButton.Height.Set(50f, 0f);
+			profileButton.VAlign = 1f;
+			profileButton.HAlign = 1f;
+			profileButton.Top.Set(-num, 0f);
+			profileButton.OnMouseOver += FadedMouseOver;
+			profileButton.OnMouseOut += FadedMouseOut;
+			profileButton.OnClick += GoLogsClick;
+			profileButton.SetSnapPoint("Profile", 0);
+			uIElement.Append(profileButton);
+			_buttonProfile = profileButton;
 			UIElement uIElement3 = MakeButton_OpenWorkshopWorldsImportMenu();
 			uIElement3.HAlign = 0f;
 			uIElement3.VAlign = 0f;
@@ -317,13 +_,16 @@
 			UILinkPoint linkPoint4 = _helper.GetLinkPoint(idRangeEndExclusive++, _buttonPublishWorlds);
 			UILinkPoint linkPoint5 = _helper.GetLinkPoint(idRangeEndExclusive++, _buttonBack);
 			UILinkPoint linkPoint6 = _helper.GetLinkPoint(idRangeEndExclusive++, _buttonLogs);
+			UILinkPoint linkPoint7 = _helper.GetLinkPoint(idRangeEndExclusive++, _buttonProfile);
 			_helper.PairLeftRight(linkPoint3, linkPoint);
 			_helper.PairLeftRight(linkPoint4, linkPoint2);
 			_helper.PairLeftRight(linkPoint5, linkPoint6);
+			_helper.PairLeftRight(linkPoint6, linkPoint7);
 			_helper.PairUpDown(linkPoint3, linkPoint4);
 			_helper.PairUpDown(linkPoint, linkPoint2);
 			_helper.PairUpDown(linkPoint4, linkPoint5);
 			_helper.PairUpDown(linkPoint2, linkPoint6);
+			_helper.PairUpDown(linkPoint2, linkPoint7);
 			_helper.MoveToVisuallyClosestPoint(num, idRangeEndExclusive);
 		}
 
